snap/validate.go

```diff
@@ -162,6 +162,22 @@ func ValidateLicense(license string) error {
 	return nil
 }
 
+func validateHooks(info *Info) error {
+	for _, hook := range info.Hooks {
+		if err := ValidateHook(hook); err != nil {
+			return err
+		}
+	}
+
+	hasDefaultConfigureHook := info.Hooks["default-configure"] != nil
```

Comment:
Will adding this check impact currently installed snaps? Say I already have a snap installed with this combination (it was allowed before right?) - what happens?

Reply:
No, because the default-configure hook feature shipped with checking in place that prevents installation within the doMount install step: https://github.com/snapcore/snapd/blob/master/overlord/snapstate/check_snap.go#L569